src/core/lib/security/credentials/ssl/ssl_credentials.cc

```diff
@@ -66,12 +68,61 @@ grpc_ssl_credentials::create_security_connector(
   absl::optional<std::string> overridden_target_name =
       args->GetOwnedString(GRPC_SSL_TARGET_NAME_OVERRIDE_ARG);
   auto* ssl_session_cache = args->GetObject<tsi::SslSessionLRUCache>();
-  grpc_core::RefCountedPtr<grpc_channel_security_connector> sc =
-      grpc_ssl_channel_security_connector_create(
-          this->Ref(), std::move(call_creds), &config_, target,
-          overridden_target_name.has_value() ? overridden_target_name->c_str()
-                                             : nullptr,
-          ssl_session_cache == nullptr ? nullptr : ssl_session_cache->c_ptr());
+  tsi_ssl_session_cache* tsi_ssl_session_cache =
+      ssl_session_cache == nullptr ? nullptr : ssl_session_cache->c_ptr();
+
+  const tsi_ssl_root_certs_store* root_store;
+  // Use default (e.g. OS) root certificates if the user did not pass any root
+  // certificates.
+  if (config_.pem_root_certs == nullptr) {
+    const char* pem_root_certs =
+        grpc_core::DefaultSslRootStore::GetPemRootCerts();
+    if (pem_root_certs == nullptr) {
+      gpr_log(GPR_ERROR, "Could not get default pem root certs.");
+      return nullptr;
+    }
+    size_t root_len = strlen(pem_root_certs);
+    char* default_roots = strcpy(new char[root_len + 1], pem_root_certs);
+    config_.pem_root_certs = default_roots;
+    root_store = grpc_core::DefaultSslRootStore::GetRootStore();
+  } else {
+    config_.pem_root_certs = config_.pem_root_certs;
+    root_store = nullptr;
+  }
+
+  grpc_core::RefCountedPtr<grpc_channel_security_connector> sc = nullptr;
```

Comment:
nit: I know lots of the C-core does this, but the normal C++ style is to use underscore-separated non-abbreviated words for variable names instead of abbreviations. To that end, can we change this variable name from `sc` to `security_connector`?

Reply:
This was copy paste from elsewhere in the code - I agree that we should switch, will make the change